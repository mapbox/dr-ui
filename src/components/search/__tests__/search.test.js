import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './search-test-cases';
import visit from 'unist-util-visit';
import SiteSearchAPIConnector from '@elastic/search-ui-site-search-connector';

describe('search', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.site.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.site;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.dark.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.dark;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.disableModal.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.disableModal;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    test('input element exists', () => {
      visit(tree, 'input', node => {
        expect(node.type).toBe('input');
      });
    });
  });

  describe(testCases.narrow.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.narrow;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    test('title element exists [a11y]', () => {
      visit(tree, 'title', node => {
        expect(node.children[0]).toBe('Search');
      });
    });
  });

  describe(testCases.withLayout.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.withLayout;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.withConnector.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.withConnector;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });

    test('uses provided connector', () => {
      expect(testCase.props.connector).toBeInstanceOf(SiteSearchAPIConnector);
      expect(testCase.props.connector.engineKey).toBe('123');
    });
  });
});
