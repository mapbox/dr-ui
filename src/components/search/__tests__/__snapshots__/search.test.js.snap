// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search Basic search with \`site\` set to show filter toggle. Open https://app.swiftype.com/engines/docs/analytics/live while developing to watch query usage. renders as expected 1`] = `
<div
  className="h36 relative"
>
  <div>
    <div>
      <button
        className="flex-parent flex-parent--center-cross btn--gray color-gray-light btn btn--stroke py3 pl6 pr12 round w-full"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="mr6 color-gray"
        >
          <svg
            className="icon w18 h18"
          >
            <use
              xlinkHref="#icon-search"
            />
          </svg>
        </span>
         
        <span
          className="color-gray"
        >
          Search
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`search Basic search. Open https://app.swiftype.com/engines/docs/analytics/live while developing to watch query usage. renders as expected 1`] = `
<div
  className="h36 relative"
>
  <div>
    <div>
      <button
        className="flex-parent flex-parent--center-cross btn--gray color-gray-light btn btn--stroke py3 pl6 pr12 round w-full"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="mr6 color-gray"
        >
          <svg
            className="icon w18 h18"
          >
            <use
              xlinkHref="#icon-search"
            />
          </svg>
        </span>
         
        <span
          className="color-gray"
        >
          Search
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`search Search with \`disableModal\` option set. Open https://app.swiftype.com/engines/docs/analytics/live while developing to watch query usage. renders as expected 1`] = `
<div
  className="h36 relative"
>
  <div>
    <div
      className="w-full"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="disabled-example-label"
        aria-owns={null}
        role="combobox"
      >
        <label
          className="cursor-pointer"
          htmlFor="disabled-example-input"
          id="disabled-example-label"
        >
          <div
            className="absolute flex-parent flex-parent--center-cross flex-parent--center-main w36 h36"
          >
            <svg
              className="icon color-gray w18 h18"
            >
              <title>
                Search
              </title>
              <use
                xlinkHref="#icon-search"
              />
            </svg>
          </div>
        </label>
        <input
          aria-activedescendant={null}
          aria-autocomplete="list"
          aria-controls={null}
          aria-labelledby="disabled-example-label"
          autoComplete="off"
          className="input bg-white px36 h36"
          id="disabled-example-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search docs.mapbox.com"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`search Search with \`narrow\` option set. Open https://app.swiftype.com/engines/docs/analytics/live while developing to watch query usage. renders as expected 1`] = `
<div
  className="h36 relative"
>
  <div>
    <div>
      <button
        className="flex-parent flex-parent--center-cross btn--gray color-gray-light btn btn--stroke py3 pl6 pr12 round"
        onClick={[Function]}
        style={
          Object {
            "paddingLeft": "12px",
            "paddingRight": "12px",
          }
        }
      >
        <span
          className="color-gray"
        >
          <svg
            className="icon w18 h18"
          >
            <title>
              Search
            </title>
            <use
              xlinkHref="#icon-search"
            />
          </svg>
        </span>
         
      </button>
    </div>
  </div>
</div>
`;

exports[`search Search with custom connector. Open https://app.swiftype.com/engines/docs/analytics/live while developing to watch query usage. renders as expected 1`] = `
<div
  className="h36 relative"
>
  <div>
    <div>
      <button
        className="flex-parent flex-parent--center-cross btn--gray color-gray-light btn btn--stroke py3 pl6 pr12 round w-full"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className="mr6 color-gray"
        >
          <svg
            className="icon w18 h18"
          >
            <use
              xlinkHref="#icon-search"
            />
          </svg>
        </span>
         
        <span
          className="color-gray"
        >
          Search
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`search Search with dark background, custom placeholder. Open https://app.swiftype.com/engines/docs/analytics/live while developing to watch query usage. renders as expected 1`] = `
<div
  className="py24 px24 bg-gray-dark"
>
  <div
    className="w-full"
  >
    <div
      className="h36 relative"
    >
      <div>
        <div>
          <button
            className="flex-parent flex-parent--center-cross btn--gray color-gray-light btn btn--stroke py3 pl6 pr12 round btn--white w-full"
            onClick={[Function]}
            style={Object {}}
          >
            <span
              className="mr6"
            >
              <svg
                className="icon w18 h18"
              >
                <use
                  xlinkHref="#icon-search"
                />
              </svg>
            </span>
             
            <span
              className=""
            >
              Search
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`search Set \`resultsOnly: true\` to true. Select an item to see search results, with a custom message when there are no search results for a particular query. Open https://app.swiftype.com/engines/docs/analytics/live while developing to watch query usage. renders as expected 1`] = `
<div>
  <select
    aria-label="Select a product"
    defaultValue=""
    onChange={[Function]}
  >
    <option
      disabled={true}
      value=""
    >
      Select a product
    </option>
    <option
      value="Studio"
    >
      Studio
    </option>
    <option
      value="Geocoding"
    >
      Geocoding
    </option>
    <option
      value="Banana"
    >
      Banana
    </option>
  </select>
  <div
    className="hmax360 scroll-auto scroll-styled"
  >
    <div
      className="relative"
    >
      <div />
    </div>
  </div>
</div>
`;
